lang/test/org/partiql/lang/eval/EvaluatingCompilerGroupByTest.kt

```diff
@@ -932,4 +953,96 @@ EvaluatingCompilerGroupByTest : EvaluatorTestBase() {
             null)
     }
 
+    @Test
+    fun missingGroupByTest() {
+        checkInputThrowingEvaluationException(
+            "SELECT MAX(@v2), @v2 FROM `[1, 2.0, 3e0, 4, 5d0]` AS v2",
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 19L,
+                    Property.BINDING_NAME to "v2"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupBySFWTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT x.title, MIN(money) as price
+            FROM books AS x, x.price AS money
+            WHERE author like '%Milan%'
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 8L,
+                    Property.BINDING_NAME to "x"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupByJoinTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT MAX(o.cost), c.firstName 
+            FROM customers AS c 
+            INNER JOIN orders AS o ON c.customerId = o.customerId
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 21L,
+                    Property.BINDING_NAME to "c"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupByHavingTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT MAX(o.cost) as total_cost, o.sellerId AS seller_id
+            FROM orders AS o
+            GROUP BY o.customerId 
+            HAVING COUNT(1) > 1
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 35L,
+                    Property.BINDING_NAME to "o"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupByOuterQueryTest() {
```

Comment:
I would add a test to make sure the error and its message is correct when the inner query uses GROUP BY and the inner queries SELECT uses a variable that is not bound by the GROUP BY clause

Reply:
Good idea. Added a new test for the inner clause